tests/test_sync_hooks_additional_dependencies.py

```diff
@@ -121,3 +121,88 @@ def test_sync_hooks_additional_dependencies(tmp_path, poetry_cwd):
         "psycopg[pool]==3.1.18",
         "types-requests==2.31.0.20240311",
     ]
+
+
+@pytest.mark.parametrize(
+    ("poetry_deps", "additional_deps", "expected_additional_deps"),
+    [
+        (["a==1", "b"], ["a"], ["a==1"]),
+        (["a==1", "b"], ["a[x]==2"], ["a==1"]),
+        (["a[x]==1", "b"], ["a"], ["a[x]==1"]),
+        (["a[x]==1", "b"], ["a[x]"], ["a[x]==1"]),
+        (["a==1", "b"], ["a == 2"], ["a==1"]),
+        (["a==1", "b"], ["a<=2"], ["a==1"]),
+        (["a==1", "b"], ["a>=1"], ["a==1"]),
+    ],
+)
+def test_sync_hook_additional_deps_with_no_new_deps(
```

Comment:
Wow, I hadn't noticed how `sync_hook_additional_deps` and `sync_hooks_additional_dependencies` were badly named. You don't have to do this, but if you want to rename either, please feel free.

Also, in test function names, I like using `__` to split the name of the function under test and the description of the test.

Reply:
No worries. I renamed for now `sync_hook_additional_deps` to `_sync_hooks_additional_dependencies` but this is probably not what you had in mind :-p Please tell me which names you prefer.

>  I like using __ to split the name [...]

Done. I also added `__` in `test__sync_hooks_additional_dependencies`, which looks a bit strange... I will change it together with the new names.

Reply:
> this is probably not what you had in mind :-p

Honnestly it's already much better than what it was. At list it's clear which one is the outer "public" one and which one is the inner. We'll find better names but it doesn't have to be in this PR so let it live this way. Thank you :)